clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -68,7 +68,7 @@
  */
 public class ConfigDef {
 
-    public static final Object NO_DEFAULT_VALUE = new String("");
+    public static final Object NO_DEFAULT_VALUE = "";
```

Comment:
This change is not right. We are using this field as a sentinel and that's why there is the `new String` there. We could perhaps just use `new Object` as this seems to confuse `findBugs` and people (it's not the first time that someone thought this was a bug).